pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PulsarChannelInitializer.java

```diff
@@ -144,7 +145,7 @@ protected void initChannel(SocketChannel ch) throws Exception {
         // auto-read.
         ch.pipeline().addLast("flowController", new FlowControlHandler());
         ServerCnx cnx = newServerCnx(pulsar, listenerName);
-        ch.pipeline().addLast("handler", cnx);
+        ch.pipeline().addLast("handler", (ChannelHandler) cnx);
```

Comment:
Sorry, I didn't get this. `ServerCnx` extends `ChannelInboundHandlerAdapter`, why is this necessary?

Reply:
Solve the error hints by the IDEA. @Jason918

Reply:
<img width="708" alt="image" src="https://user-images.githubusercontent.com/2770146/163811188-d1b8f381-0742-4cd8-bd60-b587454403d5.png">

No errors in my local IDEA, my version: IntelliJ IDEA 2021.2.4 (Ultimate Edition)

Reply:
@gaozhangmin - are you able to provide a screenshot of the error message? My IntelliJ does not report this error, either.

Reply:
<img width="1066" alt="image" src="https://user-images.githubusercontent.com/16235121/164374413-a5715d3d-a1ae-475c-b2f7-22f591a134b7.png">


I have the same issue, this is my screenshot on IntelliJ IDEA 2022.1 (Ultimate Edition)

Reply:
Thanks for sharing @nodece. I am running with IntelliJ IDEA 2021.3.3 (Ultimate Edition). I wonder why we're seeing this discrepancy.

Reply:
I also use  IntelliJ IDEA 2022.1  @michaeljmarshall  @nodece

Reply:
<img width="1066" alt="image" src="https://user-images.githubusercontent.com/18204803/165018011-7e4bcf71-4395-4d6d-9246-b133a7ae5e30.png">

It looks like a bug of Idea, both `PulsarDecoder` and my customized `Decoder` extend the same class `ChannelInboundHandlerAdapter`, while inheriting `Decoder` works well.